### ISSUE TITLE
Constraint for member named “identity” in Grails 2.5.3 generates exception when mocking for test

### ISSUE THREAD

**Issue by stenix71** [grails/grails-core#9505]
In my Grails 2.5.3 application, I have a domain class with a member named "identity". The member has a constraint looking like this:

static constraints = {
    identity maxSize: 15, nullable: false, unique: true
}
My unit test mocks the domain class like this:

@TestFor(CustomerController)
@Mock(Customer)
class CustomerControllerSpec extends Specification {
...
...but when running the test I get this exception:

groovy.lang.MissingMethodException: No signature of method: org.grails.datastore.mapping.keyvalue.mapping.config.Family.identity() is applicable for argument types: (java.util.LinkedHashMap) values: [[maxSize:15, nullable:false, unique:true]]
Possible solutions: identity(groovy.lang.Closure)

    at org.grails.datastore.mapping.config.groovy.MappingConfigurationBuilder.invokeMethod(MappingConfigurationBuilder.groovy:55)
    at com.twentyonegrams.mailportal.shipment.Customer._clinit__closure1(Customer.groovy:10)
    at com.twentyonegrams.mailportal.shipment.Customer._clinit__closure1(Customer.groovy)
    at org.grails.datastore.mapping.config.groovy.MappingConfigurationBuilder.evaluate(MappingConfigurationBuilder.groovy:83)
    at org.grails.datastore.mapping.config.AbstractGormMappingFactory.createMappedForm(AbstractGormMappingFactory.java:63)
    at org.grails.datastore.mapping.keyvalue.mapping.config.GormKeyValueMappingFactory.createMappedForm(GormKeyValueMappingFactory.java:39)
    at org.grails.datastore.mapping.keyvalue.mapping.config.GormKeyValueMappingFactory.createMappedForm(GormKeyValueMappingFactory.java:28)
    at org.grails.datastore.mapping.keyvalue.mapping.config.KeyValuePersistentEntity.<init>(KeyValuePersistentEntity.java:35)
    at org.grails.datastore.mapping.keyvalue.mapping.config.KeyValueMappingContext.createPersistentEntity(KeyValueMappingContext.java:95)
    at org.grails.datastore.mapping.model.AbstractMappingContext.addPersistentEntities(AbstractMappingContext.java:174)
    at grails.test.mixin.domain.DomainClassUnitTestMixin.mockDomains(DomainClassUnitTestMixin.groovy:86)
    at org.spockframework.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:153)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:84)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:88)
    at org.spockframework.runtime.extension.builtin.AbstractRuleInterceptor$1.evaluate(AbstractRuleInterceptor.java:37)
    at grails.test.runtime.TestRuntimeJunitAdapter$1$2.evaluate(TestRuntimeJunitAdapter.groovy:49)
    at org.spockframework.runtime.extension.builtin.TestRuleInterceptor.intercept(TestRuleInterceptor.java:38)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:87)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:88)
    at org.spockframework.runtime.extension.builtin.AbstractRuleInterceptor$1.evaluate(AbstractRuleInterceptor.java:37)
    at grails.test.runtime.TestRuntimeJunitAdapter$3$4.evaluate(TestRuntimeJunitAdapter.groovy:76)
    at org.spockframework.runtime.extension.builtin.ClassRuleInterceptor.intercept(ClassRuleInterceptor.java:38)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:87)
    at org.junit.runners.Suite.runChild(Suite.java:128)
    at org.junit.runners.Suite.runChild(Suite.java:27)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
    at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
    at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)

**Comment by droggo** [comment]
I believe this is related to https://github.com/grails/grails-core/issues/9444 . There are restricted words which cannot be used in unit tests in 2.5.x

**Comment by stenix71** [comment]
Any work-around available? I cannot change the name of the properties in this project.

**Comment by graemerocher** [comment]
Try use `delegate.identity`

**Comment by stenix71** [comment]
Tried to use delegate like this:

```
    static constraints = {
        delegate.identity( maxSize: 15, nullable: false, unique: true )
    }

```
But it did not help.

**Comment by stenix71** [comment]
Does this mean that it is fixed for the 2.5 branch as well?
